vllm/worker/enc_dec_model_runner.py

```diff
@@ -340,7 +340,11 @@ def profile_run(self) -> None:
 
         # Run the model with the dummy inputs.
         num_layers = self.model_config.get_num_layers(self.parallel_config)
-        kv_caches = [None] * num_layers
+        # use an empty tensor instead of `None`` to force Dynamo to pass
+        # it by reference, rather by specializing on the value ``None``.
+        kv_caches = [
+            torch.tensor([], dtype=torch.float32, device=self.device)
```

Comment:
Can you add that float32 is just a placeholder? Or, can you use the actual dtype instead?

Reply:
added in [35a3e44](https://github.com/vllm-project/vllm/pull/8875/commits/35a3e44231df71d7600f0356f0fd188e17021716)